lightning/src/lib.rs

```diff
@@ -44,5 +44,12 @@ pub mod ln;
 pub mod routing;
 
 mod prelude {
-	pub use alloc::{vec, vec::Vec, string::String};
-}
\ No newline at end of file
+    #[cfg(feature = "no_std")]
```

Comment:
I think we should directly check for the `hashbrown` feature. Ideally we'd be able to use `hashbrown` with `std` or `no_std`, but require it for `no_std`.

Reply:
Hrm, I tried directly naming a `hashbrown` feature, but I think it collides with the dependency name. Maybe I'm doing something wrong?

Reply:
Huh, I thought if you don't even write out the feature and just make the dependency optional you can use it as a feature in cfg.

Reply:
Ok, maybe that's what I was doing wrong. I tried writing out `hashbrown = ["hashbrown"]` as a feature. Will try what you described.